### DISCUSSION TITLE
cut image keeping size / slice image with constant shift

### DISCUSSION THREAD

**Discussion by Ratinod** [ImageMagick/ImageMagick#5568] [Help] [answered]
1) How to slice an image while keeping the original image size and keeping the position of the sliced image? (see the image below)
![question](https://user-images.githubusercontent.com/12137233/190510273-bedace22-9a65-4c1d-925a-d2082e7369e0.png)

I managed to reach only this, but it does not save the position:
`magick rose.png -crop 23x23 -alpha set -background none out.png`

2) How to slice image with constant shift / overlapping (see the image below)
![question2](https://user-images.githubusercontent.com/12137233/190512321-0d672bdd-6975-456b-937a-ecbecc965280.png)

**Comment by snibgo** [comment]
The built-in "`rose:`" image is 70x46 pixels. Here is one method to do what I think you want. Windows syntax.
```
magick -size 70x46 xc:None NULL: ( rose: -crop 70x20 ) -layers Composite x.png
```
Use any numbers you want in "`-crop`".

Does that help?

**Reply by Ratinod** [reply]
> Does that help?

Yes. Thanks! This is the answer to the first question.(I can’t imagine how much you need to delve into the documentation to get the required result)

What about the second question for image 32x32, crop:16x32 step: 8 (overlapping `-crop`)?

**Comment by snibgo** [comment, marked as answer]
You just need to have read https://imagemagick.org/script/command-line-options.php#layers , the "-layers composite" entry (and "-crop" and "-size", of course). The trick is, of course, to know that these can solve your problem.

Overlapping crops is trickier (and my brain is a bit foggy).

"-crop 4x1@" makes 4 crops, as equal as they can be. "-crop 4x1@+20+0" makes 4 crops that overlap horizontally, but by how much? Let us try it:
```
magick -size 1000x1000 xc: -crop 4x1@+20+0 info:
xc:[0] XC 265x1000 1000x1000+0+0 16-bit sRGB 0.000u 0:00.000
xc:[1] XC 265x1000 1000x1000+245+0 16-bit sRGB 0.000u 0:00.002
xc:[2] XC 265x1000 1000x1000+490+0 16-bit sRGB 0.000u 0:00.004
xc:[3] XC 265x1000 1000x1000+735+0 16-bit sRGB 0.000u 0:00.007
```
So each result is 15 pixels wider than one-quarter of 1000. The second crop starts at x=245, which is 5 pixels to the left of a non-overlapping crop. (The first crop ends at 15 pixels to the right of a non-crop, so the overlap is 5+15=20 pixels.)

The third crop starts 10 pixels to the left of a non-overlap, and the fourth and final crop starts 15 pixels to the left of a non-overlap, and finishes as we would hope at the far-right of the input.

Does that make sense? If you want an x-overlap of 20 pixels, then use an x-offset (the third number of "4x1@+20+0") of 20. I expect a y-overlap works in the same way.

(EDIT: Yes, the numbers make sense. Four crops have three areas of overlap. If each overlap is 20 pixels, then we need 3*20=60 total extra pixels from the input, compared to non-overlap. We want each crop to be the same width, so divide 60 by 4, makes an extra 15 pixels in each crop.)

For your example, this ...
```
magick -size 32x32 xc: -crop 3x1@+8+0 info:
xc:[0] XC 16x32 32x32+0+0 16-bit sRGB 0.000u 0:00.000
xc:[1] XC 16x32 32x32+8+0 16-bit sRGB 0.000u 0:00.002
xc:[2] XC 16x32 32x32+16+0 16-bit sRGB 0.000u 0:00.005
```
... does the trick.

**Reply by Ratinod** [reply]
Thanks a lot! 
For 3 days I tried to figure out how to do this by reading the documentation. But my mistake was that I hoped to find the answer in one place. [Here "Usage/crop/#crop"](https://imagemagick.org/Usage/crop/#crop)
Here is the code for the second question I was looking for:
`magick -size 32x32 xc:None NULL: ( rose32.png -crop 3x1@+8+0 ) -layers Composite x.png`
Once again, thank you so much!
